homeassistant/components/srp_energy/sensor.py

```diff
@@ -37,18 +39,22 @@ class SrpEntity(CoordinatorEntity[SRPEnergyDataUpdateCoordinator], SensorEntity)
     _attr_translation_key = "energy_usage"
 
     def __init__(
-        self, coordinator: SRPEnergyDataUpdateCoordinator, config_entry: ConfigEntry
+        self,
+        coordinator: SRPEnergyDataUpdateCoordinator,
+        config_entry: ConfigEntry,
     ) -> None:
         """Initialize the SrpEntity class."""
         super().__init__(coordinator)
         self._attr_unique_id = f"{config_entry.entry_id}_total_usage"
+
+        # Configure device
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, config_entry.entry_id)},
             name="SRP Energy",
```

Comment:
```suggestion
            name=f"SRP Energy {config_entry.title}",
```
Could append the id if you make the switch as mentioned in the config flow. For existing it would append location which should also be fine.